src/main/java/org/jabref/JabRefExecutorService.java

```diff
@@ -87,6 +79,28 @@ public boolean executeAndWait(Callable command) {
         }
     }
 
+    /**
+     * This method allows to execute a callable task that provides a return value after the calculation is done.
+     * 
+     * @param command The task to execute.
+     * @return A Future object that provides the returning value.
+     */
+    public <T> Future<T> executeAndReturn(Callable<T> command) {
+        Objects.requireNonNull(command);
+        return executorService.submit(command);
+    }
+    
+    public <T> List<Future<T>> executeAll(Collection<Callable<T>> tasks) {
+        Objects.requireNonNull(tasks);
+        List<Future<T>> futures = new ArrayList<>(); 
+        try {
+            futures =  executorService.invokeAll(tasks);
+        } catch (InterruptedException e) {
```

Comment:
Please use `logger.error` and pass the exception as an argument. Btw, Invokation is German not English ;-)

Reply:
Unfortunately, `Logger` does not provide any `PrintStream` to print the stacktrace as suggested here: https://stackoverflow.com/a/1149712 by using `e.printStackTrace(anyPrintStreamHere)`. So what's there preferred way to print the StackTrace to the LOGGER.error method?